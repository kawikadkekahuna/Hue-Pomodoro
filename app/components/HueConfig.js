// @flow
import React, { Component, PropTypes } from "react";
import { Link } from "react-router";
import styles from "./Home.css";

export default class HueConfig extends Component {
  static propTypes = {};

  componentWillMount() {
    console.log("mounted");
  }

  render() {
    return (
      <div className={styles.container} data-tid="container">
        <Link to="/">Home</Link>
        <div>
          <h1>API KEY</h1>
          <button onClick={this.props.initHueConfig}>Detect config</button>
        </div>
      </div>
    );
  }
}
