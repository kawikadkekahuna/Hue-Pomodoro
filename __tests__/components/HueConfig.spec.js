import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HueConfig from '../../app/components/HueConfig';

describe('HueConfig Component', () => {
  test('has correct default state', () => {
    const tree = shallow(<HueConfig />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
