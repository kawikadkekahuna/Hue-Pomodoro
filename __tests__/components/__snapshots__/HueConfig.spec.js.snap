// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HueConfig Component has correct default state 1`] = `
<div>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/pomodoro"
  >
    <NavigationArrowBack />
  </Link>
  <button>
    Detect config
  </button>
</div>
`;

exports[`HueConfig Component has correct state when disabled 1`] = `
<div>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/pomodoro"
  >
    <NavigationArrowBack />
  </Link>
  <button>
    Detect config
  </button>
</div>
`;
